source/credentials_provider_imds.c

```diff
@@ -322,8 +386,48 @@ static int s_imds_on_incoming_headers_fn(
     return AWS_OP_SUCCESS;
 }
 
+static void s_imds_query_token(struct aws_credentials_provider_imds_user_data *aws_credentials_provider_imds_user_data);
+static void s_imds_query_instance_role_name(struct aws_credentials_provider_imds_user_data *imds_user_data);
 static void s_imds_query_instance_role_credentials(struct aws_credentials_provider_imds_user_data *imds_user_data);
 
+static bool s_isspace(unsigned char c) {
```

Comment:
Very minor - prefer the sized types when its our own code (uint8_t)

Reply:
I tried use the original version, but have to do function pointer type conversion for the prediction argument. Then gcc will give warning.

Reply:
I mean just replace "unsigned char" with "uint8_t", as per the definition of aws_byte_predicate_fn.